tgui/packages/tgui/interfaces/TankCompressor.js

```diff
@@ -0,0 +1,316 @@
+        </Stack.Item>
+        <Stack.Item basis={0} grow>
+          <AlertBoxes
+            text_content="Tank Integrity Faltering"
+            icon_name="exclamation-triangle"
+            icon_break
+            color="yellow"
+            active={Pressure >= leakPressure}
+          />
+        </Stack.Item>
+        <Stack.Item basis={0} grow>
+          <Stack vertical fill>
+            <Stack.Item grow>
+              <AlertBoxes
+                text_content="Leak Hazard"
+                icon_name="biohazard"
+                color="red"
+                active={
+                  (Pressure >= leakPressure && Pressure < fragmentPressure)
+                  || leaking
+                }
+              />
+            </Stack.Item>
+            <Stack.Item grow>
+              <AlertBoxes
+                text_content="Explosive Hazard"
+                icon_name="bomb"
+                color="red"
+                active={Pressure >= fragmentPressure}
+              />
+            </Stack.Item>
+          </Stack>
+        </Stack.Item>
+      </Stack>
+    </Section>
+  );
+  const MainControl = (
+    <Section title="Compressor Control">
+      <Stack fill>
+        <Stack.Item grow>
+          <Slider
+            minValue={0}
+            maxValue={maxTransfer}
+            value={transferRate}
+            stepPixelSize={12.5}
+            step={0.5}
+            unit="L/S"
+            onDrag={(e, new_rate) => act('change_rate', { target: new_rate })}
+          />
+        </Stack.Item>
+        <Stack.Item>
+          <Button
+            disabled={!tankPresent || (!!leaking && Pressure < leakPressure)}
+            selected={active}
+            icon={active ? 'power-off' : 'times'}
+            onClick={() => act('toggle_injection')}>
+            {active ? 'On' : 'Off'}
+          </Button>
+        </Stack.Item>
+      </Stack>
+    </Section>
+  );
+  const GasesBreakdown = (
+    <Section fill scrollable>
+      <Stack>
+        {portData.map((individualGasmix) => (
+          <Stack.Item key={individualGasmix.ref}>
+            <Section title={individualGasmix.name}>
+              {!individualGasmix.total_moles && (
+                <Modal>{'No Gas Present'}</Modal>
+              )}
+              <GasmixParser {...individualGasmix} />
+            </Section>
+          </Stack.Item>
+        ))}
+      </Stack>
+    </Section>
+  );
+  return (
+    <>
+      <Stack.Item>{Status}</Stack.Item>
+      <Stack.Item>{MainControl}</Stack.Item>
+      <Stack.Item basis={0} grow>
+        {GasesBreakdown}
+      </Stack.Item>
+    </>
+  );
+};
+
+const TankCompressorRecords = (props, context) => {
+  const { act, data } = useBackend(context);
+  const { records = [], disk, storage } = data;
+  const [activeRecordRef, setActiveRecordRef] = useSharedState(
+    context,
+    'recordRef',
+    records[0]?.ref
+  );
+  const activeRecord = records.find((record) => activeRecordRef === record.ref);
+  const RecordBreakdown = (
```

Comment:
When doing this, you are going to run all of this code regardless of whether it ends up visible. You should inline it and run it conditionally.

Reply:
I couldn't manage to inline it but I did add a activeRecordRef check. Will this still be problematic?